lib/src/header_parser/utils.dart

```diff
@@ -326,7 +326,8 @@ class Macro {
 /// Tracks if a binding is 'seen' or not.
 class BindingsIndex {
   // Tracks if bindings are already seen, Map key is USR obtained from libclang.
-  final Map<String, Struc> _structs = {};
+  final Map<String, Compound> _structs = {};
```

Comment:
Don't we have subclasses Struc and Union for Compounds?

Reply:
Fixed this, it makes more sense to use struc/union here. I've added a cast operator to the `addStructToSeen` function to fix this.